src/python/WMCore/ReqMgr/CherryPyThreads/StatusChangeTasks.py

```diff
@@ -9,103 +9,16 @@
-            # special case for aborted workflow - aborted-completed instead of completed
-            if status == "aborted" and i == 0:
-                count += 1
-                reqmgrSvc.updateRequestStatus(wf, "aborted-completed")
-                logger.debug("%s in %s moved to %s", wf, status, "aborted-completed")
-            else:
-                for j in range(i + 1):
-                    count += 1
-                    reqmgrSvc.updateRequestStatus(wf, nextStatus[j])
-                    logger.debug("%s in %s moved to %s", wf, status, nextStatus[j])
-        logger.info("%s requests moved to new state from %s", count, status)
-    return
-
-
-def moveToCompletedForNoWQJobs(reqmgrSvc, wfStatusDict, logger):
-    """
-    Handle the case when request is aborted/rejected before elements are created in GQ
-    """
-
-    statusTransition = {"aborted": ["aborted-completed"]}
-
-    for status, nextStatusList in statusTransition.items():
-        requests = reqmgrSvc.getRequestByStatus([status], detail=False)
-        count = 0
-        for wf in requests:
-            # check whether wq elements exists for given request
-            # if not, it means
-            if wf not in wfStatusDict:
-                for nextStatus in nextStatusList:
-                    reqmgrSvc.updateRequestStatus(wf, nextStatus)
-                    count += 1
-        logger.info("Total aborted-completed: %d", count)
-
-    return
-
-
-def moveToArchived(wmstatsSvc, reqmgrSvc, archiveDelayHours, logger):
-    """
-    Handle normal-archived transition.
-    By checking AgentJobInfo status we can check whether all the agent deleted the data.
-    TODO: still need to handle the case whether agent couch is not updated for a while and agent data gets deleted
-    """
-
-    currentTime = int(time.time())
-    threshold = archiveDelayHours * 3600
-    count = 0
-
-    statusTransition = {"announced": "normal-archived",
-                        "aborted-completed": "aborted-archived",
-                        "rejected": "rejected-archived"}
-
-    outputMask = ["RequestTransition"]
-
-    for status, nextStatus in statusTransition.items():
-        inputConditon = {"RequestStatus": [status], "AgentJobInfo": "CLEANED"}
-        for reqInfo in wmstatsSvc.getFilteredActiveData(inputConditon, outputMask):
-            if reqInfo["RequestName"] and (not reqInfo["RequestTransition"] or
-                (currentTime - reqInfo["RequestTransition"][-1]["UpdateTime"]) > threshold):
-                try:
-                    reqmgrSvc.updateRequestStatus(reqInfo["RequestName"], nextStatus)
-                    count += 1
-                except Exception as ex:
-                    logger.error("Fail to update %s: %s", reqInfo["RequestName"], str(ex))
-        # convert to start status for the logging purpose
-        initStatus = "aborted" if status == "aborted-completed" else status
-        logger.info("Total %s-archived: %d", initStatus, count)
-
-    return
```

Comment:
Alan, the reason the functions are out side of class is in that way we can create the script easily and run it separately. I think I have script to use that way.  Also it is easier to test and debug. 
However, if you prefer change to class method as in the patch, this would be fine as well.